### PR TITLE
Add JSON into the design flow

### PR DESCRIPTION
**Pull request by andysun2015**
*Description of changes:*
Adds JSON into the design flow.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/doxygen/pages.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ Shadow Library Design
 
 The Device Shadow library is designed as a set of convenience macros and functions that are used to assemble and parse MQTT topic strings reserved for the Device Shadow. It does not wrap any function of the MQTT stack underneath. An application can use any MQTT library of its choice to interact with the Device Shadow service.
 
-The diagram below demonstrate how an application uses the Device Shadow library and a MQTT library to interact with the Device Shadow service.
+The diagram below demonstrate how an application uses the Device Shadow library, a MQTT library and a JSON library to interact with the Device Shadow service.
```
Comment (sarenameas) [inline docs/doxygen/pages.txt:47]:
```suggestion
The diagram below demonstrates how an application uses the Device Shadow library, a MQTT library, and a JSON library to interact with the Device Shadow service.
```

Comment (andysun2015) [reply docs/doxygen/pages.txt]: Check [this commit](https://github.com/aws/device-shadow-for-aws-iot-embedded-sdk/pull/16/commits/9fa4897ade3fc81b4a638d2be0e21d66b05ee51c) for the update.

File: docs/plantuml/shadow_design_typicaloperation.pu [hunk 2]
```diff
@@ -3,21 +3,18 @@ skinparam classFontSize 8
...
-    participant "JSON" as json
-end box
-
-box "Shadow Library" #LightBlue
-    participant "Shadow" as shadow
-end box
-
 box "Application" #LightGreen
     participant "Application" as application
     participant "callback" as callback
```
Comment (sarenameas) [inline docs/plantuml/shadow_design_typicaloperation.pu:8]: Should we capitalize Callback too since all the other life lines are?

Comment (andysun2015) [reply docs/plantuml/shadow_design_typicaloperation.pu:8]: correct! will modify it.

**Review by jainamt — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**